source/main/cardano/CardanoNode.js

```diff
@@ -300,35 +310,57 @@ export class CardanoNode {
         walletArgs,
         logStream: logFile,
         nodeImplementation,
-        stateDir: config.workingDir,
-        cluster: config.cluster,
+        stateDir: workingDir,
+        cluster,
+        block0Path,
+        block0Hash,
+        secretPath,
+        configPath,
       });
 
       this._node = node;
 
-      try {
-        await promisedCondition(() => node.connected, startupTimeout);
-        // Setup livecycle event handlers
-        node.on('message', this._handleCardanoNodeMessage);
-        node.on('exit', this._handleCardanoNodeExit);
-        node.on('error', this._handleCardanoNodeError);
-        // Request cardano-node to reply with port
-        node.send({ QueryPort: [] });
-        _log.info(
-          `CardanoNode#start: cardano-node child process spawned with PID ${node.pid}`,
-          { pid: node.pid }
-        );
-        resolve();
-      } catch (_) {
-        reject(
-          new Error('CardanoNode#start: Error while spawning cardano-node')
-        );
-      }
+      node
+        .start()
+        .then(api => {
+          _log.info(
+            `CardanoNode#start: cardano-node child process spawned with PID ${node.pid}`,
+            { pid: node.pid }
+          );
+          node.pid = 0; // TODO: Add real PID here
+          node.connected = true;
+          this._handleCardanoNodeMessage({ ReplyPort: api.requestParams.port });
```

Comment:
I assume this will be changed as part of the clean up process.

Reply:
Done!

Reply:
For now we will continue exposing just the wallet PID.